### PR TITLE
Gem Geometry fix for 81x

### PR DESCRIPTION
**Pull request by jshlee**
For GEMGeometryBuilder
- added correct sized GEMSuperChambers and GEMChambers in GEMGeometryBuilderFromDDD. Only GEMEtaPartitions were made properly before with GEMSuperChambers and GEMChambers using the same dimensions as GEMEtaPartitions (ME0 is still like this). This wasn’t an issue before as chambers were not used.
- had wrong local to global direction in half of the chambers due to odd and even chambers have different rotations
- made Geometry/GEMGeometryBuilder/data/GEMSpecsFilter.xml, this needs to be added for all geometries using GEM so that GEMGeometryBuilderFromDDD can find GEMChamber and its associated GEMEtaPartitions 

fixed GeometryExtended2015MuonGEMDev so that we can run full muon upgrade software in 81X

GEMDetId.h
- used layer = 0 for superChamber (better to have separate DetId for super chamber and chamber)
- needed 1 more bit for layer, which results in a change in the rawID
- added superChamberId() function

GEMGeometry.h 
- better superChamber() function

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by jshlee** [comment]
New superChamber and chamber size

buildSuperChamber GEMBox11S  Re -1 Ri 1 St 1 La 1 Ch 35 Ro 1
size 14.1087 24.1853 56.75 5.575
buildChamber GEMBox11S  Re -1 Ri 1 St 1 La 1 Ch 35 Ro 1
size 14.1087 24.1853 56.75 1.735

buildSuperChamber GEMBox11L  Re -1 Ri 1 St 1 La 1 Ch 34 Ro 1
size 14.1087 25.4992 64.15 5.575
buildChamber GEMBox11L  Re -1 Ri 1 St 1 La 1 Ch 34 Ro 1
size 14.1087 25.4992 64.15 1.735

buildSuperChamber GEMBox21L  Re -1 Ri 1 St 3 La 1 Ch 14 Ro 1
size 28.8539 62.55 95.55 5.575
buildChamber GEMBox21L  Re -1 Ri 1 St 3 La 1 Ch 14 Ro 1
size 28.8539 62.55 95.55 1.735

previous version had wrong unit added for SuperChamber and chamber was made form sensitive area size

**Comment by civanch** [comment]
please test

File: Configuration/Geometry/python/GeometryExtended2015MuonGEMDevReco_cff.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Ideal geometry, needed for transient ECAL alignement
 from Configuration.Geometry.GeometryExtended2015MuonGEMDev_cff import *
-from Configuration.Geometry.GeometryReco_cff import *
+from Configuration.Geometry.GeometryRecoDB_cff import *
```
Comment (ianna) [inline Configuration/Geometry/python/GeometryExtended2015MuonGEMDevReco_cff.py]: @jshlee - is it intentional that you pick up reco geometry from DB?

File: Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc [hunk 2]
```diff
@@ -34,243 +33,112 @@ GEMGeometryBuilderFromDDD::~GEMGeometryBuilderFromDDD()
 
 GEMGeometry* GEMGeometryBuilderFromDDD::build(const DDCompactView* cview, const MuonDDDConstants& muonConstants)
 {
-  std::string attribute = "ReadOutName"; // could come from .orcarc
-  std::string value     = "MuonGEMHits";    // could come from .orcarc
-  DDValue val(attribute, value, 0.0);
+  std::string attribute = "MuStructure"; // "ReadOutName"; // could come from .orcarc
```
Comment (ianna) [inline Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc]: @jshlee - ReadOutName attribute is used to convert DD sensitive detectors to G4 ones. Please, check that this change does not affect GEM sim hits.

File: Configuration/Geometry/python/GeometryExtended2015MuonGEMDevReco_cff.py (hunk 1 above)
Comment (jshlee) [inline Configuration/Geometry/python/GeometryExtended2015MuonGEMDevReco_cff.py]: @ianna - yes. using GeometryReco_cff leads to missing hcalParameters

File: Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc (hunk 2 above)
Comment (jshlee) [inline Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc]: @ianna - I change this to read from Geometry/GEMGeometryBuilder/data/GEMSpecsFilter.xml, so we can loop through chambers and then loop through the associated eta partitions.
The SensitiveDetector eta partitions are the same as the previous version (GHA10X and GHA20X).

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**